### PR TITLE
COMS-932 implement API clients logic

### PR DESCRIPTION
**Pull request by SergeyYesin**
- added publish routes
- modified routes
- modified migrations

### CODE REVIEW

File: src/Http/Requests/ApiScopesUpdate.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace CompleteSolar\ApiClients\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class ApiScopesUpdate extends FormRequest
```
Comment (yaroslav-pohil) [inline src/Http/Requests/ApiScopesUpdate.php:7]: Request isn't used

File: src/Models/ApiClient.php [hunk 2]
```diff
@@ -151,4 +151,9 @@ function (Builder $query) use ($name) {
             }
         )->get();
     }
-}
\ No newline at end of file
+
+    public function scopeFindByApiKey($query, $apiKey)
```
Comment (yaroslav-pohil) [inline src/Models/ApiClient.php]: Add param/return types. Also it's not a scope. Scope must return Builder object, not a model. Correct naming should be `public static function findByApiKey`

File: src/Models/ApiClientScope.php [hunk 3]
```diff
@@ -7,4 +7,19 @@
 class ApiClientScope extends Model
 {
     protected $fillable = ['name','description'];
+
+    public function getRouteKeyName()
```
Comment (yaroslav-pohil) [inline src/Models/ApiClientScope.php]: Please remove if not used

File: src/Models/ApiClientScope.php [hunk 4]
```diff
@@ -7,4 +7,19 @@
...
+    {
+        return 'name';
+    }
+
+    public function scopeFindByNames($query, array $names = [])
```
Comment (yaroslav-pohil) [inline src/Models/ApiClientScope.php]: Again - it's not a scope. Additionally if no scopes will be provided in the route above, I assume api client will have all scopes attached, which is wrong.
Additionally always need to set param/return types.

File: src/Models/ApiClientScope.php [hunk 5]
```diff
@@ -7,4 +7,19 @@
...
+    {
+        return $query->whereIn('name', $names)->get();
+    }
+
+    public function scopeName($query, string $name)
```
Comment (yaroslav-pohil) [inline src/Models/ApiClientScope.php]: This scope is useless. We can use `ApiClientScope::whereName('test')` instead without defining new functions.

**Review by yaroslav-pohil — CHANGES_REQUESTED** [review]

File: src/Models/ApiClientScope.php (hunk 3 above)
Comment (SergeyYesin) [reply src/Models/ApiClientScope.php]: It's used for model-route binding in triggering scope

File: src/Models/ApiClientScope.php (hunk 5 above)
Comment (SergeyYesin) [reply src/Models/ApiClientScope.php]: I prefer not to use magic that much, but ok

**Pull request merged**